### ISSUE TITLE
Better searching for hyphenated words

### ISSUE THREAD

**Issue by adia** [GrandComicsDatabase/gcd-django#114]
An indexer mentioned that searching for "Law-Breakers" finds it as a hyphenated word and as individual words but not as a combined one word (Lawbreakers).

It seems that Elasticsearch can do this using the word-delimiter token filter with the catenate_words and preserve_original parameters:

https://discuss.elastic.co/t/hyphen-search/8478/2
https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-word-delimiter-tokenfilter.html

This can index "law-breakers" as any of "law-breakers", "law breakers" and "lawbreakers", but implementing this in our search is not straighforward because the "standard" tokenizer we're using will remove the hyphens before they've reached the token filter. And there are some warnings that there might be side-effects on ranking...

We can install this Elasticsearch plugin...
https://github.com/jprante/elasticsearch-icu
... which provides the ICU tokenizer, which covers more languages than the standard tokenized and also has an option for "latin don't break on hyphens" rules.

In short, it seems there is a way to do it but it needs more complicated configuration. We need to investigate more and if we decide to do all this, the search indexes will need to be rebuilt.
